pkg/server/registry/apps/validator.go

```diff
@@ -298,6 +329,34 @@ func (s *Validator) getPermissions(ctx context.Context, image string, app *apiv1
 	return result, nil
 }
 
+func (s *Validator) getWorkloads(ctx context.Context, image string, app *apiv1.App) (map[string]v1.Container, error) {
+	result := map[string]v1.Container{}
```

Comment:
That's valid, what's the reasoning behind this best practice just so I know for the future?

Reply:
In Go, slices and maps use an array under the hood. When `append` or map assignment determines this array needs to grow, a new array array with twice the size is allocated and the data copied over. This can be expensive if done over and over, and you can save quite a bit of time/compute by allocating as much as possible upfront.

Reply:
That makes a ton of sense. Thanks for the explanation! I'll remember this for the future.

I went ahead and fixed this in the most recent push.